### PR TITLE
Fix focus highlight states for tabs and side nav

### LINKED ISSUE #3686: Tabs and sidebar links don't show active states when focused
**Describe the bug**

When an active tab or active sidebar link is focused then the active bar is hidden. This is an issue with JavaScript apps as there is no page refresh once the link is clicked, so the active state is invisible until the user clicks elsewhere on the page.

**To Reproduce**

1. Go to https://vanillaframework.io/docs/patterns/tabs
2. Scroll down to the first example.
3. Click on the active tab (Machine summary) and the underline will disappear.
4. Go to https://vanillaframework.io/docs/patterns/navigation
5. Scroll down to the side navigation example.
6. Click on the active link (Commissioning and hardware testing scripts) and the border on the left will disappear.

**Expected behavior**

The active state should be visible.

**Additional context**

It seems that these borders are being hidden so that a focus state can be shown (which doesn't seem to exist anymore on mouse click):
https://github.com/canonical-web-and-design/vanilla-framework/blob/master/scss/_patterns_tabs.scss#L79
https://github.com/canonical-web-and-design/vanilla-framework/blob/5d3821024bd80c355c12cfed33a1e2662461f976/scss/_patterns_side-navigation.scss#L203

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Fix tab highlights for focus and active so that the underline appears.
- Fix side nav highlights when focusing on the active link so that the left border appears.
- Drive-by update so the examples URL is correct in the hacking doc.

Fixes #3686.

## QA

- Go to https://vanilla-framework-3698.demos.haus/docs/examples/patterns/tabs
- Click on the active tab (Machine summary) and the underline should not disappear.
- Click on another tab, the underline should be visible for focus and active states.
- Use your keyboard to tab to the active tab, it should show the blue border and the underline should not be visible.
- Go to https://vanilla-framework-3698.demos.haus/docs/examples/patterns/side-navigation/docs
- Click on the active link (Commissioning and hardware testing scripts) and the border on the left should not disappear.
- Use your keyboard to tab to the active link, it should show the blue border and the left border should not be visible.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

**Comment by bartaz** [comment]
Thanks @huwshimi !

File: scss/_patterns_side-navigation.scss [hunk 1]
```diff
@@ -202,7 +202,7 @@
 
     // vf-highlight-bar is rendered above focus outline
     // so we need to hide it on focus
-    &:focus::before {
+    &:focus-visible::before {
```
Comment (bartaz) [inline scss/_patterns_side-navigation.scss:205]: `:focus-visible` is only supported by very latest browsers, we need to keep fallback `:focus` for the others.

Similar as we do in the mixin:
https://github.com/canonical-web-and-design/vanilla-framework/blob/57cdc27c42c892c9d834826fcd62f0e3682bfad2/scss/_global_functions.scss#L64-L77

(so you need to add focus styles both for `focus` and `focus-visible` states, but then remove them from `:focus:not(:focus-visible)`, as in the snippet above.

File: scss/_patterns_tabs.scss [hunk 2]
```diff
@@ -77,7 +77,7 @@
       }
 
       // make sure tab strip border doesn't overlap focus outline
-      &:focus {
+      &:focus-visible {
```
Comment (bartaz) [inline scss/_patterns_tabs.scss:80]: We still need `focus` fallback for older browsers (as described in other comment).

**Review by bartaz — CHANGES_REQUESTED** [review]

File: scss/_patterns_side-navigation.scss (hunk 1 above)
Comment (huwshimi) [reply scss/_patterns_side-navigation.scss]: Thanks, I've reversed these to also handle `:focus`.

File: scss/_patterns_tabs.scss (hunk 2 above)
Comment (huwshimi) [reply scss/_patterns_tabs.scss]: Fixed this one also.

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**